docs/developers/content-creation/creating-a-role.md

```diff
@@ -40,3 +44,526 @@ suicideMultiplier = -1
+```lua
+    if victim:GetSubRole() == ROLE_TESTER then return end
+  
+    if victim:GetTeam() ~= TEAM_TRAITOR then return end
+
+    if SpecDM and (victim.IsGhost() and victim:IsGhost()) then return end
+```
+
+Theoretically, you can put these 3 statements directly below each other. So that you can understand them better, I have inserted paragraphs between them. 
+Before we go any further, let's try to understand these statements now.
+
+`if victim:GetSubRole() == ROLE_TESTER then return end`
+
+“==” means something like "is equal". "~=" would mean "is unequal". Basically the line means "If the victim is tester as subrole" (if victim:GetSubRole() == ROLE_TESTER), then terminate the function (then return end). At this point it is important to access the subrole. If the base role were accessed, the tester would be a traitor.
+
+`if victim:GetTeam() ~= TEAM_TRAITOR then return end`
+
+This line means: If the victim is not a member of the traitor team (if victim:GetTeam() ~= TEAM_TRAITOR), then terminate the function. Here, again, it is important to access the Traitor Team and not the Traitor role, as we want to include other Traitor roles, such as the Vampire, for the function.
+
+`if SpecDM and (victim.IsGhost() and victim:IsGhost()) then return end`
+
+This line is something special. It shall prevent the function from being called when the players are playing Spectator Deathmatch.
+
+With these conditions, we can ensure that the subsequent code is called only if the victim belonging to the hook is a Traitor colleague.
+
+Next we want to store all players in variable. We do this by typing `local plys = player.GetAll()`. As a result, we can now access all players via the variable "plys".
+Now we have to sort out all the players who do not match what we need and when we have found a player who meets our conditions, perform the corresponding function. We can achieve this by using a "for-loop".
+You initialize a for-loop with the following code:
+
+`for i=1, #plys do`
+
+With this, we basically iterate through each player and check them, for the following conditions.
+So next we need a variable that represents the player we are currently checking. we do this by adding the following line:
+
+`local ply = plys[i]`
+
+Now when we access “ply”, we thereby access the player we are checking in the current iteration.
+Next, we want to write a condition that will only be met if the current player is alive and has the role Tester. Feel free to try developing this condition yourself. You can also use the GMOD Wiki if you get stuck. Otherwise you can scroll further and examine the solution.
+
+`if ply:GetSubRole() == ROLE_TESTER and ply:Alive() then`
+
+Next, we use an "if-else statement". Else is used for blocks of code that will be executed instead if the If-statement has not been met.
+With this block we develop the actual function of the role. This statement will define what happens to the tester himself when one of his traitor colleagues has actually died.
```

Comment:
I'm not sure if this has to be done that in depth. But I'm not against it either. Maybe rethink it?